tools/tikv-control.md

```diff
@@ -0,0 +1,164 @@
+
+## General options
+
+`tikv-ctl` provides two operation modes:
+
+- Remote mode: accept the service address of TiKV through the `--host` flag
+
+    For this mode, if SSL is started in TiKV, `tikv-ctl` also needs to specify the related certificate file. For example:
+
+    ```
+    $ tikv-ctl --ca-path ca.pem --cert-path client.pem --key-path client-key.pem --host 127.0.0.1:21060 <subcommands>
+    ```
+
+- Local mode: use the `--db` flag to specify the local TiKV data directory path
+
+Unless otherwise noted, all commands supports both the remote mode and the local mode.
+
+Besides, `tikv-ctl` has two simple commands `--to-hex` and `--to-escaped`, which are used to make simple changes to the form of the key.
+
+Generally, use the `escaped` form of the key. For example:
+
+```bash
+$ tikv-ctl --to-escaped 0xaaff
+\252\377
+$ tikv-ctl --to-hex "\252\377"
+AAFF
+```
+
+> **Note:** When you specify the `escaped` form of the key in a command line, it is required to enclose it in double quotes. Otherwise, bash eats the backslash and wrong result is returned.
+
+## Subcommands, some options and flags
+
+This section describes the subcommands that `tikv-ctl` supports in detail. Some subcommands support a lot of options. For all details, run `tikv-ctl --help <subcommand>`.
+
+### View information of the Raft state machine
+
+Use the `raft` subcommand to view the status of the Raft state machine at a specific moment. The status information includes two parts: three structs (**RegionLocalState**, **RaftLocalState**, and **RegionApplyState**) and the corresponding Entries of a certain piece of log.
+
+Use the `region` and `log` subcommands to obtain the above information respectively. The two subcommands both support the remote mode and the local mode at the same time. Their usage and output are as follows:
+
+```bash
+$ tikv-ctl --host 127.0.0.1:21060 raft region -r 2
+region id: 2
+region state key: \001\003\000\000\000\000\000\000\000\002\001
+region state: Some(region {id: 2 region_epoch {conf_ver: 3 version: 1} peers {id: 3 store_id: 1} peers {id: 5 store_id: 4} peers {id: 7 store_id: 6}})
+raft state key: \001\002\000\000\000\000\000\000\000\002\002
+raft state: Some(hard_state {term: 307 vote: 5 commit: 314617} last_index: 314617)
+apply state key: \001\002\000\000\000\000\000\000\000\002\003
+apply state: Some(applied_index: 314617 truncated_state {index: 313474 term: 151})
+```
+
+### View the Region size
+
+Use the `size` command to view the Region size:
+
+```bash
+$ tikv-ctl --db /path/to/tikv/db size -r 2
+region id: 2
+cf default region size: 799.703 MB
+cf write region size: 41.250 MB
+cf lock region size: 27616
+```
+
+### Scan to view MVCC of a specific range
+
+The `--from` and `--to` options of the `scan` command accepts two escaped forms of raw key, and uses the `--show-cf` flag to specify the column families that you need to view.
```

Comment:
The `--from` and `--to` options of the `scan` command **accept** two escaped forms of raw key, and **use** the `--show-cf` flag to specify the column families that you need to view.